doc/ikfast/ikfast_tutorial.rst

```diff
@@ -73,210 +52,83 @@ Inside your catkin workspace's ``./src`` directory: ::
 
-Or without ROS: ::
+Tweaking the creation process
+-----------------------------
 
- python /path/to/create_ikfast_moveit_plugin.py "$MYROBOT_NAME" "$PLANNING_GROUP" "$MOVEIT_IK_PLUGIN_PKG" "$BASE_LINK_NAME" "$EEF_LINK_NAME" "$IKFAST_OUTPUT_PATH"
+The process of creating the IKFast MoveIt plugin comprises several steps:
 
-Usage
------
-The IKFast plugin should function identically to the default KDL IK Solver, but with greatly increased performance. The MoveIt configuration file is automatically edited by the moveit_ikfast script but you can switch between the KDL and IKFast solvers using the *kinematics_solver* parameter in the robot's kinematics.yaml file: ::
+1. Download (and slightly modify) the docker image provided by `personalrobotics <https://hub.docker.com/r/personalrobotics/ros-openrave>`_
+2. Convert the ROS URDF file to Collada required for OpenRAVE: ::
 
- rosed "$MYROBOT_NAME"_moveit_config kinematics.yaml
+     rosrun collada_urdf urdf_to_collada <myrobot_name>.urdf <myrobot_name>.dae
 
-Edit these parts: ::
+   Sometimes floating point issues arise in converting a URDF file to Collada, which prevents OpenRAVE to find IK solutions.
+   Using a utility script, one can easily round all numbers down to n decimal places in your .dae file.
+   If the generator takes longer than, say an hour, try the following: ::
 
- <planning_group>:
-   kinematics_solver: <myrobot_name>_<planning_group>_kinematics/IKFastKinematicsPlugin
- -INSTEAD OF-
-   kinematics_solver: kdl_kinematics_plugin/KDLKinematicsPlugin
+     rosrun moveit_ikfast round_collada_numbers.py <input_dae> <output_dae> <decimal places>
 
-Test the Plugin
-^^^^^^^^^^^^^^^
+   From experience we recommend 5 decimal places, but if the OpenRave ikfast generator takes to long to find a solution, lowering the number of decimal places should help. For example: ::
 
-Use the MoveIt RViz Motion Planning Plugin and use the interactive markers to see if correct IK Solutions are found.
+     rosrun moveit_ikfast round_collada_numbers.py <myrobot_name>.dae <myrobot_name>.rounded.dae 5
 
-Updating the Plugin
--------------------
+3. Run OpenRAVE IKFast tool to generate C++ solver code
+4. Create the MoveIt IKFast plugin package wrapping the generated solver
 
-If any future changes occur with MoveIt! or IKFast, you might need to re-generate this plugin using our scripts. To allow you to easily do this, a bash script is automatically created in the root of your IKFast package, named *update_ikfast_plugin.sh*. This does the same thing you did manually earlier, but uses the IKFast solution header file that is copied into the ROS package.
+To manually tweak the process, you can start the script ``auto_create_ikfast_moveit_plugin.sh`` from any intermediate step by providing the corresponding output from the previous step as input (``.dae`` or ``.cpp``) instead of the ``.urdf`` file.
```

Comment:
```suggestion
The ``auto_create_ikfast_moveit_plugin.sh`` script reads in the file extension from the input to determine which steps to run. To re-run the script from any intermediate step, simply provide the corresponding output from the previous step as input (``.dae`` or ``.cpp``) instead of the ``.urdf`` file.
```

Reply:
OK. Slightly modified.